### PR TITLE
Zoom status indicators

### PR DESCRIPTION
**Pull request by na9da**
### What this PR does

- [x] Adds a toast and zoom button spinner when map is zooming similar to the attached design.
- [x] Modifies the `zoomTo()` call to return a promise that resolves when the zoom animation is complete.
- [x] Implement slide in/out animation for the toast
- [x] Fixed toast positioning on mobile - shown 40px from bottom left.

image

### Testing

* Visit http://ci.terria.io/zoom-status-indicators/#share=s-hQ3zR006Fur8ObS3fZh1fiveM4V
* Click ideal zoom for the first catalog item. 
* The zoom button should show a spinner and a toast should appear at bottom center of the screen
* The zoom button will be disabled while the zoom is in progress
* Click ideal zoom for the second catalog item
* You should see the same results

### Checklist

-   ~[ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)~
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/ReactViews/AnimatedSpinnerIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import styled from "styled-components";
```
Comment (na9da) [inline lib/ReactViews/AnimatedSpinnerIcon.tsx:1]: Extracted from `Loader.tsx`

File: lib/ReactViews/Workbench/Controls/ViewingControls.jsx [hunk 2]
```diff
@@ -113,17 +120,22 @@ const ViewingControls = observer(
...
-      let zoomToView = item;
-      if (
-        item.rectangle !== undefined &&
-        item.rectangle.east - item.rectangle.west >= 360
-      ) {
-        zoomToView = this.props.viewState.terria.mainViewer.homeCamera;
-        console.log("Extent is wider than world so using homeCamera.");
-      }
-      viewer.zoomTo(zoomToView);
+      runInAction(() => {
```
Comment (na9da) [inline lib/ReactViews/Workbench/Controls/ViewingControls.jsx]: Just wraps in `runInAction`

File: lib/ReactViews/AnimatedSpinnerIcon.tsx (hunk 1 above)
Comment (zoran995) [inline lib/ReactViews/AnimatedSpinnerIcon.tsx]: It might be better to move this to `lib/Styled`

File: lib/ReactViews/Loader.tsx [hunk 3]
```diff
@@ -1,9 +1,8 @@
 import { TFunction } from "i18next";
```
Comment (zoran995) [inline lib/ReactViews/Loader.tsx:1]: Also might be better to move this to `lib/Styled`

Comment (na9da) [reply lib/ReactViews/Loader.tsx:1]: I'm a bit confused about that. How do we now decide what goes in `lib/Styled`?

Comment (zoran995) [reply lib/ReactViews/Loader.tsx:1]: Not sure, but I see `Loader` as a component that is reused in combination with other components and doesn't stand alone. It just might be easier to find it later when needed.

**Review by tephenavies — APPROVED** [review]

**Pull request merged**